text/0162-refactoring-support.md

```diff
@@ -0,0 +1,829 @@
+    [≡] AWS::Lambda::Function MyStack/Function/Resource -> Service/Function/Resource
+
+You can also refactor the resources as part of a deployment, by running `cdk 
+deploy --refactoring-action=refactor --unstable=refactor`. You will be shown the
+same table as above, but a different set of options to choose from:
+
+    ? What do you want to do? (Use arrow keys)  
+    ❯ Execute the refactor and deploy  
+      Deploy without refactoring (will cause resource replacement)  
+      Quit
+
+A few things to note about this feature:
+
+- If you pass any filters to the `deploy` command, the refactor will work on
+  those stacks plus any other stacks the refactor touches. For example, if you
+  choose to only deploy stack A, and a resource was moved from stack A to stack
+  B, the refactor will involve both stacks. But if there was a rename in, let's
+  say, stack C, it will not be refactored. The same set of filters is available
+  for the `refactor` command.
+- To perform refactoring, the CLI needs new permissions in the bootstrap stack.
+  Before using this feature, run `cdk bootstrap` for every target environment,
+  to add these new permissions.
+- A given CDK application can have multiple stacks, for different environments.
+  In that case, the CLI will group the stacks by environment and perform the
+  refactoring separately in each one. So, although you can move resources
+  between stacks, both stacks involved in the move must be in the same
+  environment. Trying to move resources across environments will result in an
+  error.
+
+### Explicit mapping
+
+#### Refactor file
+
+The previous section describes the case in which the CLI automatically computes
+the mappings, by comparing the current and proposed states. But if you don't
+want this automatic detection to happen, you can provide your own mapping, via a
+_refactor file_.
+
+The refactor file contains a map of current locations to new locations. When
+provided, this file serves as an allow-list: instead of computing the mapping,
+it will try to perform all the refactors in the list, and nothing else. The keys
+refer to existing locations; their corresponding values refer to the new
+locations they should be moved to. Both keys and values should follow the
+pattern `"<stack name>.<logical ID>"`:
+
+    {
+      "StackA.OldName": "StackB.NewName",
+      "StackC.Foo": "StackC.Bar"
+    }
+
+You can pass this mapping to the CLI via the `--refactor-mapping` option to both
+`deploy` and `refactor` commands.
+
+This is a regular JSON file, that you can write yourself. But to make it
+convenient for you, every time the CLI computes a mapping, it will automatically
+generate a refactor file at the root of your CDK application, with a timestamp
+in the name. Example: `.refactor-2025-04-03_12-05-49.json`. You can use these
+files as a starting point to edit the mapping, combine multiple mappings into
+one, split mappings into multiple files, etc.
+
+#### Skip file
```

Comment:
Isn't this better in one file together with the refactorings?

Reply:
That was my initial idea. But that file would be used to:

1. Override the result of a computed mapping (the skip part).
2. Preclude mapping computation altogether (the refactor part).

These two functions are mutually exclusive. It would be confusing to have one file with both.